### PR TITLE
Create README

### PR DESCRIPTION
**Pull request by 0xfuturistic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,160 @@
...
+[MultiStaker](https://github.com/0xfuturistic/MultiStaker/blob/main/src/MultiStaker.sol) is a smart contract designed to manage ERC20 token deposits and facilitate staking across various protocols and chains. It provides a flexible architecture that allows for both possessory and non-possessory staking, supports multiple deposits per user, and integrates cross-chain operations for L1-L2 interactions.
+
+[The contract](https://github.com/0xfuturistic/MultiStaker/blob/main/src/MultiStaker.sol) allows users to:
+
+
+- Stake deposits across various protocols through a single contract, reducing complexity and risk.
+- Retain and delegate voting rights while staking, solving the "governance lockup" problem prevalent in many DeFi protocols.
+- Stake on L1 while earning rewards on L2.
+- Choose between possessory and non-possessory staking options, to maximizing compatibility with existing protocols.
+- Stake into simultaneous slashable stakers.
```
Comment (josojo) [inline README.md]: Should it be? Simultaneously stake into slashable stakers?

File: README.md [hunk 2]
```diff
@@ -1 +1,160 @@
...
+
+The lifecycle of a deposit in the `MultiStaker` contract can be visualized as follows:
+
+
+```mermaid
+stateDiagram-v2
+    state fork_state <<fork>>
+        [*] --> fork_state
+        fork_state --> Unstaked: deposit()
+        Unstaked --> fork_state: withdraw()
```
Comment (josojo) [inline README.md:56]: I would rename Unstaked with Staked(1). THen it fits better to the Staked(N)

**Review by josojo — COMMENTED** [review]
Nice diagrams

File: README.md (hunk 1 above)
Comment (0xfuturistic) [reply README.md]: yes. you can definitely be staked into several slashable (non-possessory) stakers at the same time. I will make the distinction about non-possessory more clear!

File: README.md (hunk 2 above)
Comment (0xfuturistic) [reply README.md:56]: if `N` indicates the number of deposits, why replace `Unstaked` by `Staked(1)` when `N=0` at that time?

Comment (josojo) [reply README.md:56]: I was missreading it. Yeah makes perfect sense as it is

**Review by josojo — APPROVED** [review]

**Pull request merged**